scripts/syschdemd.nix

```diff
@@ -0,0 +1,54 @@
+{ 
+  stdenvNoCC, 
+  makeWrapper, 
+  lib, 
+
+  coreutils, 
+  daemonize,
+  glibc,
+  gnugrep,
+  systemd, 
+  util-linux, 
+  
+  defaultUser,
+  defaultUserHome
+}:
+let
+  writeShellScript = { name, src, path, ... }@args:
+    stdenvNoCC.mkDerivation ({
+      inherit name src;
+
+      dontUnpack = true;
+
+      nativeBuildInputs = [ makeWrapper ];
+
+      installPhase = ''
+        install -Dm755 $src $out/bin/${name}
+        substituteAllInPlace $out/bin/${name}
+        wrapProgram $out/bin/${name} --set PATH ${lib.escapeShellArg path}
+      '';
+    } // args);
+  
+  wrapper = writeShellScript {
+    name = "nixos-wsl-systemd-wrapper";
+    src = ./wrapper.sh;
+    path = lib.makeSearchPath "" [ 
+      "/run/wrappers/bin"  # mount
+      "${systemd}/lib/systemd"  # systemd
+    ];
+  };
+in writeShellScript {
+  name = "syschdemd";
+  src = ./syschdemd.sh;
+  path = lib.makeBinPath [ 
+    "/run/wrappers"  # mount
+    coreutils
+    daemonize
+    glibc  # getent
+    gnugrep
+    systemd  # machinectl
+    util-linux  # nsenter
+    wrapper
+  ];
+  inherit defaultUser defaultUserHome;
+}
```

Comment:
Missing a final new line which is important that git can generate diffs properly.